src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -47,16 +47,17 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
         var thunks = endpoints.Select((endpoint, _) => $$"""
             [{{StaticRouteHandlerModelEmitter.EmitSourceKey(endpoint)}}] = (
-           (del, builder) =>
+           (methodInfo, options) =>
```

Comment:
What do you think of explicitly specifying the lambda types here and the build func?

Output from source generators I've seen is always very conservative. Explicit types might reduce the chances of any conflicts or ambiguous method calls in the future.

Reply:
Is that really any different from any other public API we ship? The customer might be calling code and we might add overloads. I'm not convinced we should change anything.

Reply:
I'm thinking about the situation where the generator code is somehow used on a newer framework (no, I don't know how that could happen. Throwing out a what-if situation)

The difference is the customer controls their code and can fix it. If a generator outputs code that doesn't compile, then it's not immediately obvious what has gone wrong and what someone should do.

My point of view is generated code is always ugly. It's not like we lose anything by being explicit.

Reply:
I was motivated to avoid hardcoding too many things here under the assumption that the generated code is going to change quite a bit over the next preview. This aspect of it might not change but I valued making it easy to modify before we did any hardening to it. I think there's a lot of value in generated code that is easier to read especially as we spin things up.